src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1851 @@ interface MPSMatrixMultiplication {
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronPReLU")]
+	[DisableDefaultCtor]
+	interface MPSCnnNeuronPReLU {
+		[Export ("initWithDevice:a:count:")]
+		[Internal, Sealed]
+		IntPtr InitWith (IMTLDevice device, IntPtr /* float* */ a, nuint count);
+
+		// inlining ctor from base class
+		[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronHardSigmoid")]
+	interface MPSCnnNeuronHardSigmoid {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+
+		[Export ("initWithDevice:a:b:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a, float b);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronSoftPlus")]
+	interface MPSCnnNeuronSoftPlus {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+
+		[Export ("initWithDevice:a:b:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a, float b);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronSoftSign")]
+	interface MPSCnnNeuronSoftSign {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronELU")]
+	interface MPSCnnNeuronELU {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("initWithDevice:a:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnNeuron), Name = "MPSCNNNeuronReLUN")]
+	interface MPSCnnNeuronReLUN {
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+
+		[Export ("initWithDevice:a:b:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, float a, float b);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnConvolutionDescriptor), Name = "MPSCNNSubPixelConvolutionDescriptor")]
+	interface MPSCnnSubPixelConvolutionDescriptor {
+		[Export ("subPixelScaleFactor")]
+		nuint SubPixelScaleFactor { get; set; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSCnnConvolutionDescriptor), Name = "MPSCNNDepthWiseConvolutionDescriptor")]
+	interface MPSCnnDepthWiseConvolutionDescriptor {
+		[Export ("channelMultiplier")]
+		nuint ChannelMultiplier { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSState), Name = "MPSCNNConvolutionState")]
+	interface MPSCnnConvolutionState : MPSImageSizeEncodingState {
```

Comment:
base type has a DisableDefaultCtor
unless mentioned (comment needed) it's not clear the subclass has an `init` ?!?